.github/workflows/ci.yml

```diff
@@ -1,11 +1,229 @@
+          - { target: thumbv7m-none-eabi,              os: ubuntu-latest,                    std: 1 }
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: ./.github/actions/setup-rust
+
+      - name: Set up Docker Buildx
+        if: runner.os == 'Linux'
+        uses: docker/setup-buildx-action@v1
+      - name: Expose GitHub runtime
+        uses: crazy-max/ghaction-github-runtime@v1.0.0
+      - name: Build and push Docker image
+        id: build-docker-image
+        if: runner.os == 'Linux'
+        timeout-minutes: 60
+        run: ./build-docker-image.sh "${TARGET}"
+        env:
+          TARGET: ${{ matrix.target }}
+        shell: bash
+      - name: Set Docker image for test
+        if: steps.build-docker-image.outputs.image
+        run: |
+          TARGET_VAR="cross_target_${TARGET//-/_}_image"
+          echo "${TARGET_VAR^^}=${IMAGE}" | tee -a "${GITHUB_ENV}"
+        env:
+          TARGET: ${{ matrix.target }}
+          IMAGE: ${{ steps.build-docker-image.outputs.image }}
+        shell: bash
+      - name: Test
+        if: github.ref == 'refs/heads/staging' || github.ref == 'refs/heads/trying'
+        run: ./ci/test.sh
+        env:
+          TARGET: ${{ matrix.target }}
+          CPP: ${{ matrix.cpp }}
+          DYLIB: ${{ matrix.dylib }}
+          STD: ${{ matrix.std }}
+          RUN: ${{ matrix.run }}
+          RUNNERS: ${{ matrix.runners }}
+        shell: bash
+
+      - name: Install cross
+        if: matrix.deploy
+        run: cargo install --path . --force
+
+      - uses: ./.github/actions/cargo-install-upload-artifacts
+        if: matrix.deploy
+        with:
+          target: ${{ matrix.target }}
+
+      - name: Login to GitHub Container Registry
+        if: runner.os == 'Linux'
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Push image to GitHub Container Registry
+        if: runner.os == 'Linux' && (github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/tags/v'))
+        run: ./build-docker-image.sh --push "${TARGET}"
+        env:
+          TARGET: ${{ matrix.target }}
+        shell: bash
+
+  publish:
+    needs: [docker-image]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/setup-rust
+      - uses: ./.github/actions/cargo-publish
+        with:
+          cargo-registry-token: ${{ secrets.CARGO_REGISTRY_TOKEN }}
+
+  conclusion:
+    needs: [shellcheck, fmt, clippy, test, docker-image, publish]
+    if: always()
+    runs-on: ubuntu-latest
+    steps:
+      - name: Result
+        run: |
+          jq -C <<< "${needs}"
+
+          # Check if all needs were successfull or skipped.
+          "$(jq -r 'all(.result as $result | (["success", "skipped"] | contains([$result])))' <<< "${needs}")"
```

Comment:
Yeah, this could be made a bit more explicit, but it works as expected. The only job which is skipped currently is the docker job on the `pull_request` event. The only other way a job is skipped is when a previous fails, so in that case there will be at least one failed job, so the `all` will be false.